src/name/resolve.js

```diff
@@ -125,13 +118,46 @@ module.exports = (createCommon, options) => {
             ipfs.name.resolve(keyId, resolveOptions, (err, res) => {
               expect(err).to.not.exist()
               expect(res).to.exist()
-              expect(res.path).to.equal(`/ipfs/${value}`)
+              expect(res).to.equal(`/ipfs/${value}`)
 
               done()
             })
           })
         })
       })
     })
+
+    it('should resolve /ipns/<hash>/remainder/file.txt', async () => {
+      const add = await ipfs.add(Buffer.from('fake remainder'))
+      await ipfs.name.publish(add[0].path, { 'allow-offline': true })
+
+      const resolve = await ipfs.name.resolve(`/ipns/${nodeId}/remainder/file.txt`)
+
+      return expect(resolve).to.match(/\/ipfs\/.+\/remainder\/file.txt$/)
+    })
+
+    it('should resolve ipfs.io', async () => {
+      const r = await ipfs.name.resolve('ipfs.io', { recursive: false })
+      return expect(r).to.match(/\/ipns\/.+$/)
+    })
+
+    it('should resolve ipfs.io with remainder', async () => {
+      const r = await ipfs.name.resolve('/ipns/ipfs.io/images/ipfs-logo.svg', { recursive: false })
+      return expect(r).to.match(/\/ipns\/.+\/images\/ipfs-logo.svg$/)
+    })
+
+    it('should resolve /ipns/ipfs.io recursive', async () => {
+      const r = await ipfs.name.resolve('ipfs.io', { recursive: true })
+      return expect(r).to.match(/\/ipfs\/.+$/)
+    })
+
+    it('should resolve /ipns/ipfs.io recursive with remainder', async () => {
```

Comment:
@hugomrdias we are missing a test like this one (a recursive resolve of `/ipns/<fqdn>/some/file`), but with a path that is inside of a HAMT-sharded root. 

I believe `/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html` can be used for test: it is a safe path that will not disappear and Wikipedia is also an important consumer of sharding feature (`/wiki/` is a sharded directory).

FYSA js-ipfs may not support sharding correctly until https://github.com/ipfs/js-ipfs-http-response/pull/19 is merged first.